src/lib/components/Card/Card.tsx

```diff
@@ -57,14 +59,36 @@ export const Card: FC<CardProps> = ({
       )}
       {...theirProps}
     >
-      {imgSrc && (
-        <img
-          alt={imgAlt ?? ''}
-          src={imgSrc}
-          className={classNames(theme.img.base, theme.img.horizontal[horizontal ? 'on' : 'off'])}
-        />
-      )}
+      <Image {...props} />
       <div className={theme.root.children}>{children}</div>
     </Component>
   );
 };
+
+const Image: FC<CardProps> = ({ theme: customTheme = {}, ...props }) => {
+  const theme = mergeDeep(useTheme().theme.card, customTheme);
+  if (props.renderImage) {
+    return props.renderImage();
+  }
+  if (props.imgSrc) {
+    return (
+      <img
+        alt={props.imgAlt ?? ''}
+        src={props.imgSrc}
+        className={classNames(theme.img.base, theme.img.horizontal[props.horizontal ? 'on' : 'off'])}
```

Comment:
This is the theme that I'm talking about. :)

Reply:
But you should have access to the theme, and you can decide if you will use it or not.

Reply:
Yes, just do:

```
renderImage = () => {
  const theme = useTheme()
  ....
}

Reply:
TBH, I don't see the problem to inject, and I don't see the benefits to call the hook again.

I got the point that "one unified way to access the theme", and it seems a pretty reasonable argument, but since we are using the `renderImage` to render something exclusively inside the `Card` component it kind of loses its point. I already have the theme object available inside the component where I'm trying to render an "expected" component. So, properties that can be used in this inner component should be injected by the main component.

Also, it will make the code a little bit clean... Also consider that, if you are rendering a list of cards, and then you have 10 cards in this list. You are going to call the hooks 20 times. If you inject, you will have only half (not ideal yet, but, still... half).

Reply:
The theme is always present, it is part of the core:

https://github.com/themesberg/flowbite-react/blob/89d58dc03727d150e73ec753748eb98640bc84a3/src/lib/components/Card/Card.tsx#L46

Here is the interface for the card image theme:

https://github.com/themesberg/flowbite-react/blob/89d58dc03727d150e73ec753748eb98640bc84a3/src/lib/components/Card/Card.tsx#L20

Reply:
To the `renderImage` param you need to pass: `renderImage(theme.image)`

Reply:
Ops... Sorry:

```ts
interface RenderImage {
    (theme: FlowbiteCardImageTheme) => JSX.Element
}
```

Reply:
You may also need to pass one extra property, let me check...